compiler/rustc_ast/src/token.rs

```diff
@@ -404,7 +404,7 @@ impl Token {
         [DotDot, DotDotDot, DotDotEq].contains(&self.kind)
     }
 
-    pub fn is_op(&self) -> bool {
+    pub fn is_punct(&self) -> bool {
```

Comment:
`Punct` in proc macros is about punctuation characters (single-character), but `op` here is about operators (possibly multi-character, like `+=`).
But I'm also not sure how right is to call e.g. `#` an operator.
So renaming this is probably fine.